java/runtime/src/main/java/org/ray/runtime/WorkerContext.java

```diff
@@ -25,19 +31,53 @@
   /**
    * How many puts have been done by current task.
    */
-  private int currentTaskPutCount;
+  private AtomicInteger currentTaskPutCount;
 
   /**
    * How many calls have been done by current task.
    */
-  private int currentTaskCallCount;
+  private AtomicInteger currentTaskCallCount;
+
+  /**
+   * The ID of main thread which created the worker context.
+   */
+  private long mainThreadId;
+  /**
+   * If the multi-threading warning message has been logged.
+   */
+  private AtomicBoolean multiThreadingWarned;
 
   public WorkerContext(WorkerMode workerMode, UniqueId driverId) {
     workerId = workerMode == WorkerMode.DRIVER ? driverId : UniqueId.randomId();
-    currentTaskPutCount = 0;
-    currentTaskCallCount = 0;
+    currentTaskPutCount = new AtomicInteger(0);
```

Comment:
Should these also be reset to 0 in `setCurrentTask`?

Reply:
yep, @jovan-wong can you fix this?

Reply:
em, I don't think we should reset these to 0 in `setCurrentTask()`, because `currentTask` denotes the task was executing in a worker,  and the `currentTaskCallCount ` denotes how many times this worker invoke `Ray.call()`.